thrust/functional.h

```diff
@@ -1224,10 +1393,27 @@ struct project1st
 
   /*! Function call operator. The return value is <tt>lhs</tt>.
    */
-  __host__ __device__ const T1 &operator()(const T1 &lhs, const T2 & /*rhs*/) const {return lhs;}
+  __host__ __device__
+  constexpr const T1 &operator()(const T1 &lhs, const T2 & /*rhs*/) const
+  {
+    return lhs;
+  }
 }; // end project1st
 
-/*! \p project2nd is a function object that takes two arguments and returns 
+template <>
+struct project1st<void, void>
+{
+  __thrust_exec_check_disable__
+  template <typename T1, typename T2>
+  __host__ __device__
+  constexpr auto operator()(T1&& t1, T2&&) const
+    noexcept(noexcept(THRUST_FWD(t1))) -> decltype(THRUST_FWD(t1))
```

Comment:
`noexcept(THRUST_FWD(t1))` is always true, and `decltype(THRUST_FWD(t1))` is `decltype(t1)` which is `T1&&`.  But there is also a lot of good in consistency with other similar classes even if that results in extra typing.  So I am not requesting any changes.